.github/workflows/main.yaml

```diff
@@ -5,65 +5,128 @@ on:
+    outputs:
+      short_git_sha: ${{ steps.vars.outputs.SHORT_GIT_SHA }}
+      short_git_version_string: ${{ steps.vars.outputs.SHORT_GIT_VERSION_STRING }}
     steps:
-      - name: Checkout PX4-msgs
-        uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set short git commit SHA
+        id: vars
+        run: |
+          calculatedSha=$(git rev-parse --short ${{ github.sha }})
+          calculatedGitVersionString=$(git log --date=format:%Y%m%d --pretty=~git%cd.%h -n 1)
+          echo "SHORT_GIT_SHA=$calculatedSha" >> $GITHUB_OUTPUT
+          echo "SHORT_GIT_VERSION_STRING=$calculatedGitVersionString" >> $GITHUB_OUTPUT
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
         with:
-          path: PX4-msgs
-      - name: Run px4-msgs docker build
+          platforms: ${{ matrix.architecture }}
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          config: .github/buildkitd.toml
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Container metadata for tii-px4-msgs
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/tiiuae/tii-px4-msgs
+          tags: |
+            type=ref,event=branch,suffix=-${{ matrix.architecture }}
+            type=ref,event=pr,suffix=-${{ matrix.architecture }}
+            type=semver,pattern={{raw}},suffix=-${{ matrix.architecture }}
+            type=sha,suffix=-${{ matrix.architecture }}
+            type=raw,value=latest,enable=${{ endsWith(github.ref, github.event.repository.default_branch) }}
+
+      - name: Build and push debian package container
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          context: .
+          file: ./Dockerfile
+          # cache-from: type=gha
+          # cache-to: type=gha,mode=max
+          platforms: linux/${{ matrix.architecture }}
+          build-args: |
+            PACKAGE_NAME=px4-msgs
+            GIT_RUN_NUMBER=${{ github.run_number }}
+            GIT_COMMIT_HASH=${{ github.sha }}
+            GIT_VERSION_STRING=${{ steps.vars.outputs.SHORT_GIT_VERSION_STRING }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          provenance: false
+
+      - name: Create container and copy deb packages
         run: |
-          set -eux
-          mkdir bin
-          pushd PX4-msgs
-          ./build.sh ../bin/
-          popd
-      - name: Upload build
-        uses: actions/upload-artifact@v2
+          docker create -ti --name "px4-msgs-build-temp-${{ matrix.architecture }}" --platform=linux/${{ matrix.architecture }} $(echo "${{ steps.meta.outputs.tags }}" | head -1) bash
```

Comment:
Should we consider use ORAS to store the Debian package?
I don't know if on the consumer side makes it more difficult to use if the artifact is stored as OCI image.

Reply:
I don't know how we can manage this with multiarch images. It may be difficult or adding extra logic using tags with -arm64 or -amd64 suffixes.

Reply:
As these images are mainly to be used for development purposes, I figured the simple docker images would suffice. This would make the users just use the docker commands to fetch the deb packages to local pc or in the builds with Dockerfile. I assume the oras fetching needs a bit more elaborate steps to get this done, along with some more dependencies.

But it is a good point. We might possibly add the ORAS manifest pushing along with the docker images if needed.